typescript/agentkit/src/action-providers/near/nearActionProvider.ts

```diff
@@ -0,0 +1,150 @@
+import { z } from "zod";
+import { toBytes } from "viem";
+import { getTransactionLastResult } from "@near-js/utils";
+
+import { Network, NEAR_PROTOCOL_FAMILY, NEAR_NETWORK_ID } from "../../network";
+import { NEARWalletProvider } from "../../wallet-providers/nearWalletProvider";
+import { ActionProvider } from "../actionProvider";
+import { CreateAction } from "../actionDecorator";
+
+import {
+  GET_CROSS_CHAIN_ADDRESS_DESCRIPTION,
+  GET_CROSS_CHAIN_PUBLIC_KEY_DESCRIPTION,
+  SIGN_PAYLOAD_DESCRIPTION,
+} from "./descriptions";
+import {
+  generateAddress,
+  deriveChildPublicKey,
+  getRootPublicKey,
+  getMpcAccountIdByNetwork,
+  MpcContract,
+  AddressType,
+  SignArgs,
+} from "./utils";
+import {
+  GetCrossChainAddressInput,
+  GetCrossChainPublicKeyInput,
+  SignPayloadInput,
+} from "./schemas";
+import { DEFAULT_KEY_VERSION, DEFAULT_PATH, SUPPORTED_NETWORKS } from "./constants";
+import { MPCSignature } from "./types";
+
+/**
+ * The NearActionProvider class provides actions for the NEAR protocol family
+ */
+export class NearActionProvider extends ActionProvider<NEARWalletProvider> {
+  /**
+   * Creates an instance of NearActionProvider
+   */
+  constructor() {
+    super("near", []);
+  }
+
+  /**
+   * Returns the cross chain address for the given account id, network id, path and address type
+   *
+   * @param walletProvider - The wallet provider
+   * @param args - The get cross chain address input arguments
+   *
+   * @returns The cross chain address
+   */
+  @CreateAction({
+    name: "get_cross_chain_address",
+    description: GET_CROSS_CHAIN_ADDRESS_DESCRIPTION,
```

Comment:
Please inline the action descriptions. I appreciate the refactor, however the pattern throughout is to have these descriptions live close to the action